File: py/plugins/anthropic/src/genkit/plugins/anthropic/model_info.py
# Copyright 2025 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
# SPDX-License-Identifier: Apache-2.0

"""Anthropic Models for Genkit."""

from genkit.types import (
    Constrained,
    ModelInfo,
    Supports,
)

# Model definitions
CLAUDE_3_HAIKU = ModelInfo(
    label='Anthropic - Claude 3 Haiku',
    versions=['claude-3-haiku-20240307'],
    supports=Supports(
        multiturn=True,
        media=True,
        tools=True,
        system_role=True,
    ),
)

CLAUDE_3_5_HAIKU = ModelInfo(
    label='Anthropic - Claude 3.5 Haiku',
    versions=['claude-3-5-haiku-20241022'],
    supports=Supports(
        multiturn=True,
        media=True,
        tools=True,
        system_role=True,
    ),
)

CLAUDE_SONNET_4 = ModelInfo(
    label='Anthropic - Claude Sonnet 4',
    versions=['claude-sonnet-4-20250514'],
    supports=Supports(
        multiturn=True,
        media=True,
        tools=True,
        system_role=True,
    ),
)

CLAUDE_OPUS_4 = ModelInfo(
    label='Anthropic - Claude Opus 4',
    versions=['claude-opus-4-20250514'],
    supports=Supports(
        multiturn=True,
        media=True,
        tools=True,
        system_role=True,
    ),
)

CLAUDE_SONNET_4_5 = ModelInfo(
    label='Anthropic - Claude Sonnet 4.5',
    versions=['claude-sonnet-4-5-20250929'],
    supports=Supports(
        multiturn=True,
        media=True,
        tools=True,
        system_role=True,
        output=['text', 'json'],
        constrained=Constrained.ALL,
    ),
)

CLAUDE_HAIKU_4_5 = ModelInfo(
    label='Anthropic - Claude Haiku 4.5',
    versions=['claude-haiku-4-5-20251001'],
    supports=Supports(
        multiturn=True,
        media=True,
        tools=True,
        system_role=True,
        output=['text', 'json'],
        constrained=Constrained.ALL,
    ),
)

CLAUDE_OPUS_4_1 = ModelInfo(
    label='Anthropic - Claude Opus 4.1',
    versions=['claude-opus-4-1-20250805'],
    supports=Supports(
        multiturn=True,
        media=True,
        tools=True,
        system_role=True,
        output=['text', 'json'],
        constrained=Constrained.ALL,
    ),
)

CLAUDE_OPUS_4_5 = ModelInfo(
    label='Anthropic - Claude Opus 4.5',
    versions=['claude-opus-4-5-20251101'],
    supports=Supports(
        multiturn=True,
        media=True,
        tools=True,
        system_role=True,
        output=['text', 'json'],
        constrained=Constrained.ALL,
    ),
)

SUPPORTED_ANTHROPIC_MODELS: dict[str, ModelInfo] = {
    'claude-3-haiku': CLAUDE_3_HAIKU,
    'claude-3-5-haiku': CLAUDE_3_5_HAIKU,
    'claude-sonnet-4': CLAUDE_SONNET_4,
    'claude-opus-4': CLAUDE_OPUS_4,
    'claude-sonnet-4-5': CLAUDE_SONNET_4_5,
    'claude-haiku-4-5': CLAUDE_HAIKU_4_5,
    'claude-opus-4-1': CLAUDE_OPUS_4_1,
    'claude-opus-4-5': CLAUDE_OPUS_4_5,
}

DEFAULT_SUPPORTS = Supports(
    multiturn=True,
    media=True,
    tools=True,
    system_role=True,
    output=['text'],
)


def get_model_info(name: str) -> ModelInfo:
    """Get model info for a given model name.

    Args:
        name: Model name.

    Returns:
        Model information.
    """
    return SUPPORTED_ANTHROPIC_MODELS.get(
        name,
        ModelInfo(
            label=f'Anthropic - {name}',
            supports=DEFAULT_SUPPORTS,
        ),
    )

File: java/ai/src/main/java/com/google/genkit/ai/ModelInfo.java
/*
 * Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package com.google.genkit.ai;

import java.util.List;
import java.util.Set;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * ModelInfo contains metadata about a model's capabilities.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ModelInfo {

  @JsonProperty("label")
  private String label;

  @JsonProperty("supports")
  private ModelCapabilities supports;

  @JsonProperty("versions")
  private List<String> versions;

  /**
   * Default constructor.
   */
  public ModelInfo() {
  }

  // Getters and setters

  public String getLabel() {
    return label;
  }

  public void setLabel(String label) {
    this.label = label;
  }

  public ModelCapabilities getSupports() {
    return supports;
  }

  public void setSupports(ModelCapabilities supports) {
    this.supports = supports;
  }

  public List<String> getVersions() {
    return versions;
  }

  public void setVersions(List<String> versions) {
    this.versions = versions;
  }

  /**
   * ModelCapabilities describes what a model can do.
   */
  @JsonInclude(JsonInclude.Include.NON_NULL)
  public static class ModelCapabilities {

    @JsonProperty("multiturn")
    private Boolean multiturn;

    @JsonProperty("media")
    private Boolean media;

    @JsonProperty("tools")
    private Boolean tools;

    @JsonProperty("systemRole")
    private Boolean systemRole;

    @JsonProperty("output")
    private Set<String> output;

    @JsonProperty("context")
    private Boolean context;

    @JsonProperty("contextCaching")
    private Boolean contextCaching;

    // Getters and setters

    public Boolean getMultiturn() {
      return multiturn;
    }

    public void setMultiturn(Boolean multiturn) {
      this.multiturn = multiturn;
    }

    public Boolean getMedia() {
      return media;
    }

    public void setMedia(Boolean media) {
      this.media = media;
    }

    public Boolean getTools() {
      return tools;
    }

    public void setTools(Boolean tools) {
      this.tools = tools;
    }

    public Boolean getSystemRole() {
      return systemRole;
    }

    public void setSystemRole(Boolean systemRole) {
      this.systemRole = systemRole;
    }

    public Set<String> getOutput() {
      return output;
    }

    public void setOutput(Set<String> output) {
      this.output = output;
    }

    public Boolean getContext() {
      return context;
    }

    public void setContext(Boolean context) {
      this.context = context;
    }

    public Boolean getContextCaching() {
      return contextCaching;
    }

    public void setContextCaching(Boolean contextCaching) {
      this.contextCaching = contextCaching;
    }
  }
}
